File: Examples/VisualBasic/NoteBook/AddChildNode.vb
Imports System.Drawing
Imports System.IO
Imports System.Collections.Generic
Imports Aspose.Note

Namespace WorkingWithNoteBook
    Public Class AddChildNode
        Public Shared Sub Run()
            ' ExStart:AddChildNode
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_NoteBook()

            ' Load a OneNote Notebook
            Dim notebook = New Notebook(dataDir & Convert.ToString("Notizbuch öffnen.onetoc2"))

            ' Append a new child to the Notebook
            notebook.AppendChild(New Document(dataDir & Convert.ToString("Neuer Abschnitt 1.one")))

            dataDir = dataDir & Convert.ToString("AddChildNode_out.onetoc2")

            ' Save the Notebook
            notebook.Save(dataDir)
            ' ExEnd:AddChildNode
            Console.WriteLine(Convert.ToString(vbLf & "Child node added successfully." & vbLf & "File saved at ") & dataDir)
        End Sub
    End Class
End Namespace

File: Examples/CSharp/NoteBook/AddChildNode.cs
using System.IO;
using Aspose.Note;
using System;
using System.Collections.Generic;

namespace Aspose.Note.Examples.CSharp.WorkingWithNoteBook
{
    public class AddChildNode
    {
        public static void Run()
        {
            // ExStart:AddChildNode
            // ExFor:Notebook
            // ExFor:Document
            // ExSummary:Shows how to add new section to a notebook.

            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_NoteBook();

            // Load a OneNote Notebook
            var notebook = new Notebook(dataDir + "Notizbuch öffnen.onetoc2");

            // Append a new child to the Notebook
            notebook.AppendChild(new Document(dataDir + "Neuer Abschnitt 1.one"));

            dataDir = dataDir + "AddChildNode_out.onetoc2";

            // Save the Notebook
            notebook.Save(dataDir);

            // ExEnd:AddChildNode

            Console.WriteLine("\nChild node added successfully.\nFile saved at " + dataDir);
        }
    }
}